File: wondrous.c
// Jonathan Wong, z3460680
// 25/03/16
// Prints sequence of Wondrous numbers

#include <stdlib.h>
#include <stdio.h>
#include <assert.h>

int printWondrous (int start);

int main (int argc, char* argv []) {

    return EXIT_SUCCESS;
}

int printWondrous(int start) {
   int loop = 6;

   while (start != 1) {
       if ((start & 1) == 0) {
          start = start/2;
       }
       else {
       (start = (start * 3) + 1);
       }
       printf("%d ", start);
       loop++;
   }
   return loop;
}

File: wondrous.py
def printWondrous (start):
    count = 1
    while start != 1:
        if (start % 2) == 0:
            start = start/2
        else:
            start = start * 3 + 1
        print (str(start),end=" ")
        count = count + 1
    return count

start = int(input('Enter start number:'))
print (str(start),end=" ")
printWondrous(start)

